src/inject.js

```diff
@@ -1,120 +1,184 @@
                     stop = 1
-                    innerStreak = streakFromTo
-                }
-            }
-            if (!currentStreak && nonContributingDays > 1 && todayValue && !hasNoCurrentStreak) {
-                const lastCommit = new Date(day.attributes["data-date"].value)
-                const diff = Math.abs((new Date()).getTime() - lastCommit.getTime())
-                let unit = "days"
-                let timeDiff = Math.ceil(diff / (1000 * 3600 * 24)) - 1
-                console.log(timeDiff)
-                if (timeDiff === 1) {
-                    unit = "day"
                 }
-                if (timeDiff > 30) {
-                    unit = "month"
-                    if (timeDiff > 60) {
-                        unit = "months"
-                    }
-                    timeDiff = (new Date()).getMonth() - lastCommit.getMonth() + (12 * ((new Date()).getFullYear() - lastCommit.getFullYear()))
-                }
-                hasNoCurrentStreak = `Last contributed <time>${timeDiff} ${unit} ago</time>`
-            }
-            if (todayValue) {
-                ++evaluatingStreak
-            } else {
-                evaluatingStreak = 0
             }
-            if (evaluatingStreak > longestStreak) {
-                longestStreak = evaluatingStreak
-                longestStreakStartingDate = new Date(day.attributes["data-date"].value)
-                longestStreakEndingDate = new Date(day.attributes["data-date"].value)
-                longestStreakEndingDate.setDate(longestStreakEndingDate.getDate() + longestStreak - 1)
+
+            // if (!currentStreak && nonContributingStreak > 1 && contributionCount && !lastContributionText) {
+            if (!currentStreak && nonContributingStreak > 1 && !lastContributionText) {
+                lastContributionText = getLastContributionText(contributionDate)
             }
+
         })
 
-        if (firstCommitDate) {
-            const monthFrom = monthNames[firstCommitDate.getMonth()]
-            const dayFrom = firstCommitDate.getUTCDate()
-            const yearFrom = firstCommitDate.getFullYear()
-            const monthTo = monthNames[lastCommitDate.getMonth()]
-            const dayTo = lastCommitDate.getUTCDate()
-            const yearTo = lastCommitDate.getFullYear()
-            let currentStreakSkeleton = ""
-            if (hasNoCurrentStreak) {
-                currentStreakSkeleton = `<span class="text-muted">${hasNoCurrentStreak}</span>`
-            } else {
-                currentStreakSkeleton = `<span class="text-muted">${innerStreak}</span>`
+        if (firstContributionDate) {
+            currentStreakText = lastContributionText ? lastContributionText : currentStreakText
+            longestStreakText = getLongestStreakText(longestStreakStartingDate, longestStreakEndingDate)
+            totalContributionsText = getTotalContributionsText(firstContributionDate, lastContributionDate)
```

Comment:
these are not required to to be associated to a `var`; just for the sake of consistency in the first pass. will be removed in next PR.